developer-workflow/c-api.rst

```diff
@@ -134,6 +145,101 @@ When moving existing tests, feel free to replace ``TestError`` with
+Users of this tier may need to change their code with every minor release.
+
+In many ways, this tier is like the general C API:
+
+- it's available when ``Python.h`` is included normally,
+- it should be defined  in ``Include/cpython/``,
+- it requires tests, so we don't break it unintentionally
+- it requires docs, so both we and the users
+  can agree on the expected behavior,
+- it is tested and documented in the same way.
+
+The differences are:
+
+- Names of functions structs, macros, etc. start with the ``PyUnstable_``
+  prefix. This defines what's in the unstable tier.
+- The unstable API can change in minor versions, without any deprecation
+  period.
+- A stability note is appears in the docs.
+  This happens automatically, based on the name
+  (via :cpy-file:`Doc/tools/extensions/c_annotations.py`).
+
+Despite being “unstable”, there are rules to make sure third-party code can
+use this API reliably:
+
+* Changes and removals can be done in minor releases (3.x.0, including
+  Alphas and Betas for 3.x.0).
+* Adding new unstable API *for an existing feature* is allowed even after
+  Beta feature freeze, up until the first Release Candidate.
+  Consensus on Core Development Discourse or is needed in the Beta period.
+* Backwards-incompatible changes should make existing C callers fail to compile.
+  For example, arguments should be added/removed, or a function should be
+  renamed.
+* When moving API into or out of the Unstable tier, the old name
+  should continue to be available (but deprecated) until an incompatible
+  change is made. In other words, while we're allowed to break calling code,
+  we shouldn't break it *unnecessarily*.
+
+
+Moving API from the public tier to unstable
+-------------------------------------------
+
+* Expose the API under its new name, with the ``PyUnstable_`` prefix.
+* Make the old name an alias (e.g. a ``static inline`` function calling the
+  new function).
+* Deprecate the old name, typically using ``Py_DEPRECATED``.
+* Announce the change in What's New.
+
+The old name should continue to be available until an incompatible change is
+made. Per Python’s backwards compatibility policy (:pep:`387`),
+this deprecation needs to last at least two releases (modulo SC exceptions).
+
+The rules are relaxed for API that was introduced before the official
+Unstable tier.
+For the following, you can make an incompatible change (and remove the old name)
+as if the function was already part of the Unstable tier:
+
+* API introduced before Python 3.12 that is documented to be less stable than default.
+* API introduced before Python 3.12 that was named with a leading underscore.
+
+Moving API from the private tier to unstable
+--------------------------------------------
+
+* Expose the API under its new name, with the ``PyUnstable_`` prefix.
+* If the old name is documented, or widely used externally,
+  make it an alias and deprecate it (typically with ``Py_DEPRECATED``).
```

Comment:
```suggestion
  make it an alias and deprecate it (typically with :c:macro:`Py_DEPRECATED`).
```

Likewise, cross-reference and provide a helpful link to this macro (and if it changes, a warning will be issued)